src/zenml/integrations/mlflow/mlflow_utils.py

```diff
@@ -38,126 +46,59 @@ def setup_mlflow(
     set_experiment(experiment_name)
 
 
-def enable_mlflow_init(
-    original_init: Callable[[BasePipeline, BaseStep, Any], None],
-    experiment: Optional[str] = None,
-) -> Callable[..., None]:
-    """Outer decorator function for extending the __init__ method for pipelines
-    that should be run using mlflow
-
-    Args:
-        original_init: The __init__ method that should be extended
-        experiment: The users chosen experiment name to use for mlflow
-
-    Returns:
-        the inner decorator which extends the __init__ method
-    """
-
-    def inner_decorator(
-        self: BasePipeline, *args: BaseStep, **kwargs: Any
-    ) -> None:
-        """Inner decorator overwriting the pipeline __init__
-        Makes sure mlflow is properly set up and all mlflow logging takes place
-        within one mlflow experiment that is associated with the pipeline
-        """
-        original_init(self, *args, **kwargs)
-        setup_mlflow(
-            backend_store_uri=local_mlflow_backend(),
-            experiment_name=experiment if experiment else self.name,
-        )
-
-    return inner_decorator
-
-
-def enable_mlflow_run(run: Callable[..., Any]) -> Callable[..., Any]:
-    """Outer decorator function for extending the run method for pipelines
-    that should be run using mlflow
-
-    Args:
-        run: The run method that should be extended
-
-    Returns:
-        the inner decorator which extends the run method
-    """
-
-    def inner_decorator(
-        self: BasePipeline, run_name: Optional[str] = None
-    ) -> Any:
-        """Inner decorator used to extend the run method of a pipeline.
-        This ensures each pipeline run is run within a different mlflow context.
-
-        Args:
-            self: self of the original pipeline class
-            run_name: Optional name for the run.
-        """
-        with mlflow.start_run(run_name=run_name):
-            run(self, run_name)
-
-    return inner_decorator
-
-
-def enable_mlflow(
-    _pipeline: Type[BasePipeline], experiment_name: Optional[str] = None
-) -> Type[BasePipeline]:
-    """Outer decorator function for the creation of a ZenML pipeline with mlflow
-    tracking enabled.
+def get_or_create_mlflow_run(experiment_name: str, run_name: str) -> ActiveRun:
+    """Get or create an MLflow ActiveRun object for the given experiment and
+    run name.
 
-    In order for a pipeline to run within the context of mlflow, the mlflow
-    experiment should be associated with the pipeline directly. Each separate
-    pipeline run needs to be associated directly with a mlflow experiment. For
-    this, the __init__ and run methods need to be extended accordingly.
+    IMPORTANT: this function is not race condition proof. If two or more
+    processes call it at the same time and with the same arguments, it could
+    lead to a situation where two or more MLflow runs with the same name
+    and different IDs are created.
```

Comment:
Might be worth creating a todo or directly creating a ticket to look into this. Feel free to resolve this if its already done